internal/server/statvar/fetcher/svg_fetcher.go

```diff
@@ -0,0 +1,176 @@
+
+	pb "github.com/datacommonsorg/mixer/internal/proto"
+	"github.com/datacommonsorg/mixer/internal/server/statvar/hierarchy"
+	"github.com/datacommonsorg/mixer/internal/store"
+	"github.com/datacommonsorg/mixer/internal/store/bigtable"
+	"google.golang.org/protobuf/proto"
+)
+
+// FetchAllSVG fetches entire SVG from storage
+func FetchAllSVG(
+	ctx context.Context,
+	store *store.Store,
+) (map[string]*pb.StatVarGroupNode, error) {
+	result := map[string]*pb.StatVarGroupNode{}
+	if store.BtGroup != nil {
+		// Read stat var group cache from the allowed import group table.
+		btDataList, err := bigtable.ReadWithFilter(
+			ctx,
+			store.BtGroup,
+			bigtable.BtStatVarGroup,
+			[][]string{{""}},
+			func(jsonRaw []byte) (interface{}, error) {
+				var svgResp pb.StatVarGroups
+				if err := proto.Unmarshal(jsonRaw, &svgResp); err != nil {
+					return nil, err
+				}
+				return &svgResp, nil
+			},
+			// Only use svg from "frequent", "experimental" and custom import groups.
+			// These two import groups have the latest and wanted sv/svgs. We don't
+			// want to include those in "infrequent" etc that may have stale sv/svg.
+			func(t *bigtable.Table) bool {
+				return (strings.HasPrefix(t.Name(), "frequent") ||
+					strings.HasPrefix(t.Name(), "experimental") ||
+					t.IsCustom())
+			},
+		)
+		if err != nil {
+			return nil, err
+		}
+		// Loop through import group by order. The stat var group is preferred from
+		// a higher ranked import group.
+		var customRootNode *pb.StatVarGroupNode
+		for _, btData := range btDataList {
+			for _, row := range btData {
+				svgData, ok := row.Data.(*pb.StatVarGroups)
+				if ok && len(svgData.StatVarGroups) > 0 {
+					for k, v := range svgData.StatVarGroups {
+						if k == hierarchy.CustomSvgRoot && customRootNode == nil {
+							customRootNode = v
+						}
+						if _, ok := result[k]; !ok {
+							result[k] = v
+						} else {
+							// Merge all SVGs regardless of the import group rank as one SVG
+							// can exist in multiple import group.
+							hierarchy.MergeSVGNodes(result[k], v)
+						}
+					}
+				}
+			}
+		}
+		if customRootNode != nil {
+			customRootExist := false
+			// If custom schema is built together with base schema, then it is
+			// already in the child stat var group of "dc/g/Root".
+			for _, x := range result[hierarchy.SvgRoot].ChildStatVarGroups {
+				if x.Id == hierarchy.CustomSvgRoot {
+					customRootExist = true
+					break
+				}
+			}
+			// Populate dc/g/Custom_Root as children of dc/g/Root
+			if !customRootExist {
+				result[hierarchy.SvgRoot].ChildStatVarGroups = append(
+					result[hierarchy.SvgRoot].ChildStatVarGroups,
+					&pb.StatVarGroupNode_ChildSVG{
+						Id:                hierarchy.CustomSvgRoot,
+						SpecializedEntity: customRootNode.AbsoluteName,
+					},
+				)
+			}
+		}
+	}
+	if store.SQLClient != nil {
+		// Query for all the stat var group node
+		query :=
+			`
+					SELECT t1.subject_id, t2.object_value, t3.object_id
```

Comment:
At some point we should consolidate all sql calls and BT calls into their respective modules. Currently it's spread out across multiple modules requiring them to be knowledgable of the underlying db model and making refactoring expensive.

Can be done in a separate PR but just noting it here.

Reply:
Yes, there is a module for that. Need to move the rest code there. I can do that when touching them in a smaller PR.